scipy/optimize/tests/test_quadratic_assignment.py

```diff
@@ -69,13 +68,19 @@ def test_accuracy_1(self):
              [0, 2, 0, 2],
              [0, 1, 2, 0]]
 
-        res = quadratic_assignment(A, B, method=self.method,
-                                   options={"rng": 0, "maximize": False})
+        with assert_warns(FutureWarning):
+            res = quadratic_assignment(A, B, method=self.method,
+                                       options={"rng": 0, "maximize": False})
+
         assert_equal(res.fun, 10)
         assert_equal(res.col_ind, np.array([1, 2, 3, 0]))
 
-        res = quadratic_assignment(A, B, method=self.method,
-                                   options={"rng": 0, "maximize": True})
+        res = quadratic_assignment(
+            A,
+            B,
+            method=self.method,
+            options={"rng": default_rng(1234), "maximize": True}
+        )
```

Comment:
If you don't mind, I will push a commit to minimize the diff:
```suggestion
        rng = np.random.default_rng(745287425872459)
        res = quadratic_assignment(A, B, method=self.method,
                                   options={"rng": rng, "maximize": True})
```

Reply:
I prefer the style I had it in (it's more black-like), but don't mind if you make a change.